### ISSUE TITLE
Padded_batch with pre- or post-padding option

### ISSUE THREAD

**Issue by shahriar49** [tensorflow/tensorflow#33465]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.12.2
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
I have a dataset of variable-length sequences to feed an LSTM network and I want to try and compare pre- and post-padding in the batches, but current padded_batch function only pads at the sequences end.

**Who will benefit with this feature?**
I heard that pre-padding may be better for LSTM than post-padding, because post-padding may 'wash' the cell states before final decision making. Is it true?

**Any Other info.**

**Comment by gowthamkpr** [comment]
@shahriar49 This already exists in the later versions of tensorflow 

`tf.keras.preprocessing.sequence.pad_sequences(
    sequences,
    maxlen=None,
    dtype='int32',
    padding='pre',
    truncating='pre',
    value=0.0
)`

You can use `padding = 'pre' or 'post'` based on your use case.

**Comment by shahriar49** [comment]
@gowthamkpr 
I knew it but I don't know how to apply this function to dataset batch processor. The padded_batch function in tensorflow does both padding and batching, and it will find the required paddding size per batch dynamically. How can I implement this myself? My code right now is like this, and I am reading multiple TFRecord files and interleave them to make my mixed dataset:

```
featuresDict = {'data': tf.FixedLenFeature([], dtype=tf.string),
                'rows': tf.FixedLenFeature([], dtype=tf.int64),
                'label': tf.FixedLenFeature([], dtype=tf.int64)
               }

def parse_tfrecord(example):
    features = tf.parse_single_example(example, featuresDict)
    label = tf.one_hot(features['label'],N)
    rows = features['rows']
    data = tf.decode_raw(features['data'], tf.int64)
    data = tf.reshape(data, (rows,num_features)
    return data, label

def read_datasets(pattern, numFiles, numEpochs=None, batchSize=None):
    files = tf.data.Dataset.list_files(pattern)

    def _parse(x):
        x = tf.data.TFRecordDataset(x, compression_type='GZIP')
        return x

    dataset = files.interleave(_parse, cycle_length=numFiles, block_length=1).map(parse_tfrecord)
    padded_shapes = (tf.TensorShape([None, num_features]), tf.TensorShape([N,])))
    dataset = dataset.padded_batch(batchSize, padded_shapes)
    dataset = dataset.prefetch(buffer_size=batchSize)
    dataset = dataset.repeat(numEpochs)
    return dataset

```

**Comment by shahriar49** [comment]
@gowthamkpr
Could you please help me with this issue?

**Comment by gowthamkpr** [comment]
@shahriar49 Please post this question in [stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow) as this platform is only for bug/performance, docs or build/installation issues. Thanks!

**Comment by shahriar49** [comment]
@gowthamkpr 
I tried it but nobody give me an answer on stackoverflow. Can you help me with it?

**Comment by shahriar49** [comment]
@gowthamkpr
I tried it but nobody give me an answer on stackoverflow. Can you help me with it?

**Comment by titaneric** [comment]
Hi @shahriar49 ,

I have answered your question on [stackoverflow ](https://stackoverflow.com/questions/58525480/padded-batch-with-pre-or-post-padding-option/59212858#59212858), please check it.
